File: 0055-jump-game/0055-jump-game.py
class Solution(object):
    def canJump(self, nums):
        f = 0
        for i in range(len(nums)):
            if i > f: 
                return False
            f = max(f, i + nums[i])  
            if f >= len(nums) - 1:  
                return True
        return False 

File: 0055-jump-game/0055-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
          int x =0;
          for(int i=0;i<nums.length;i++){
                if(x<i){
                    return false;
                }
                x=Math.max(x,i+nums[i]);
          }
          return true;
    }
}